src/nvim/tui/tui.c

```diff
@@ -709,6 +729,22 @@ static void update_attrs(TUIData *tui, int attr_id)
     }
   }
 
+  if ((attrs.url.size != tui->url.size) || (!strnequal(attrs.url.data, tui->url.data,
+                                                       tui->url.size))) {
+    if (tui->url.data != NULL) {
+      out(tui, S_LEN("\x1b]8;;\x1b\\"));
+    }
+
+    if (attrs.url.data != NULL) {
+      StringBuilder sb = KV_INITIAL_VALUE;
+      kv_printf(sb, "\x1b]8;;%s\x1b\\", attrs.url.data);
+      out(tui, sb.items, sb.size);
+      kv_destroy(sb);
+    }
```

Comment:
If we are starting a new URL the old URL (if it exists) does not need to be terminated. It works just like any other SGR sequence (e.g. you don't need to "terminate" one color before starting another).

Reply:
8c5830471b691681e4f636a277bdce7539bd2edd